### PR TITLE
[MM-19589] Update the Jira plugin for Go 1.13

### PR DESCRIPTION
**Pull request by cpoile**
#### Summary
- Have to update dependencies  to satisfy the stricter go 1.13 requirements

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -31,9 +31,11 @@ require (
 )
 
 // Workaround for https://github.com/golang/go/issues/30831 and fallout.
-replace github.com/golang/lint => github.com/golang/lint v0.0.0-20190227174305-8f45f776aaf1
+replace github.com/golang/lint => github.com/golang/lint v0.0.0-20181217174547-8f45f776aaf1
```
Comment (crspeller) [inline go.mod]: I think some these might be able to be removed. Have you tried that?

Comment (cpoile) [reply go.mod]: Good idea. Tried it, here's what's required now.

File: go.mod [hunk 2]
```diff
@@ -30,10 +30,8 @@ require (
 	google.golang.org/grpc v1.21.1 // indirect
 )
 
-// Workaround for https://github.com/golang/go/issues/30831 and fallout.
-replace github.com/golang/lint => github.com/golang/lint v0.0.0-20190227174305-8f45f776aaf1
-
 // Workaround for Jira GDPR change. Remove this and update golang.org/x/oauth2 when it's fixed.
-replace golang.org/x/oauth2 => github.com/cpoile/oauth2 v0.0.0-20190717235208-10945bd
+// See: https://github.com/golang/oauth2/issues/312
+replace golang.org/x/oauth2 => github.com/cpoile/oauth2 v0.0.0-20190717234427-10945bd219c7
```
Comment (mickmister) [inline go.mod]: @cpoile I see you're still famous :)

**Comment by DHaussermann** [comment]
@cpoile please advise if there is any functional testing required for this PR.

**Comment by cpoile** [comment]
@DHaussermann nope. If it compiles, the PR works.

**Pull request merged**